### PR TITLE
Update auth to work with python 3.8

### PR DESCRIPTION
**Pull request by EngHabu**
Signed-off-by: Haytham Abuelfutuh <haytham@afutuh.com>

# TL;DR
Fixing multiprocessing stuff for auth for 3.8.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
* Use explicit fork context when creating the Process, had to move the queue around.
* Added better loggers.
* Deprecated the old scope in favor of a new scope since it needs to be a list.
* Cleaned up imports

## Tracking Issue
NA

### CODE REVIEW

File: flytekit/clis/auth/auth.py [hunk 1]
```diff
@@ -39,6 +24,7 @@
 # Identifies the key used for storing and fetching from keyring. In our case, instead of a username as the keyring docs
 # suggest, we are storing a user's oidc.
 _keyring_access_token_storage_key = "access_token"
+_keyring_id_token_storage_key = "id_token"
```
Comment (EngHabu) [inline flytekit/clis/auth/auth.py]: Not needed, right?
```suggestion
```

File: flytekit/clis/auth/auth.py [hunk 2]
```diff
@@ -233,10 +243,18 @@ def _initialize_credentials(self, auth_token_resp):
...
         self._params.update(
-            {"code": auth_code.code, "code_verifier": self._code_verifier, "grant_type": "authorization_code"}
+            {
+                "code": auth_code.code,
+                "code_verifier": self._code_verifier,
+                "grant_type": "authorization_code",
+            }
         )
+
+        if self._client_secret:
```
Comment (EngHabu) [inline flytekit/clis/auth/auth.py:255]: So this wasn't here before... is it needed?

File: flytekit/clients/raw.py [hunk 3]
```diff
@@ -27,8 +27,11 @@ def _refresh_credentials_standard(flyte_client):
     :return:
     """
 
-    _credentials_access.get_client(flyte_client.url).refresh_access_token()
-    flyte_client.set_access_token(_credentials_access.get_client(flyte_client.url).credentials.access_token)
+    client = _credentials_access.get_client(flyte_client.url)
+    if client.can_refresh_token:
+        client.refresh_access_token()
```
[the remark this answers is not in the document]
Comment (EngHabu) [reply flytekit/clients/raw.py:33]: As long as refresh token remains optional?

File: flytekit/clients/raw.py [hunk 4]
```diff
@@ -27,8 +28,32 @@ def _refresh_credentials_standard(flyte_client):
...
+    Merge the scope value between the old scope config option and the new list option.
+
+    :return: The scope to use for basic auth flow
+    """
+    deprecated_single_scope = _DEPRECATED_SCOPE.get()
+    if deprecated_single_scope:
+        return deprecated_single_scope
+    oauth_scopes = OAUTH_SCOPES.get()
+    scope = oauth_scopes[0]  # there's a default so this is okay to do
+    if len(oauth_scopes) != 1:
```
Comment (EngHabu) [inline flytekit/clients/raw.py:49]: Is this true?

**Review by EngHabu — COMMENTED** [review]
LGTM.. one question

**Review by wild-endeavor — APPROVED** [review]

**Pull request merged**